Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -0,0 +1,213 @@
+                    await message.ModifyAsync(async m =>
+                        m.Embed = (await BuildPollEmbedAsync(campaign, campaign.ApproveCount, campaign.OpposeCount)).Build());
+                }
+            }
+        }
+
+        private async Task CreatePromoDialog(PromotionActionCreationData campaign)
+        {
+            if (!await _designatedChannelService.AnyDesignatedChannelAsync(campaign.GuildId,
+                DesignatedChannelType.PromotionInterface))
+                return;
+
+            var pollChannel = await GetPollChannelAsync(_discordSocketClient.GetGuild(campaign.GuildId));
+
+            var campaignSummary = await _promotionCampaignRepository.GetCampaignSummaryByIdAsync(campaign.Campaign.Id);
+
+            var message = await pollChannel.SendMessageAsync(embed: (await BuildPollEmbedAsync(campaignSummary, 1, 0)).Build());
+
+            await message.AddReactionsAsync(new[] {(IEmote)_approveEmoji, _disapproveEmoji});
+
+            SetDialogCache( new CachedPromoDialog
+            {
+                MessageId = message.Id,
+                CampaignId = campaignSummary.Id,
+            });
+
+            await _promotionDialogRepository.CreateAsync(message.Id, campaign.Campaign.Id);
+        }
+
+        private async Task DeletePromoDialog(PromotionActionCreationData campaign)
+        {
+            if (!_memoryCache.TryGetValue(GetKey(campaign.Campaign.Id), out CachedPromoDialog poll))
+                return;
+            var pollChannel = await GetPollChannelAsync(_discordSocketClient.GetGuild(campaign.GuildId));
+
+            var message =  await pollChannel.GetMessageAsync(poll.MessageId);
+
+            if (!await _promotionDialogRepository.TryDeleteAsync(message.Id))
+                throw new InvalidOperationException("Dialog to be deleted does not exist");
+
+            _memoryCache.Remove(GetKey(poll.CampaignId));
+            _memoryCache.Remove(GetKey(poll.MessageId));
+            await message.DeleteAsync();
+        }
+
+        private async Task<EmbedBuilder> BuildPollEmbedAsync(PromotionCampaignSummary campaign, int Approve, int Oppose, string error = null)
+        {
+                var boldRole = $"**{MentionUtils.MentionRole(campaign.TargetRole.Id)}**";
+                var user = _discordSocketClient.GetUser(campaign.Subject.Id);
+
+                ValueTask<Color> aviColor;
+
+                if ((user.GetAvatarUrl(size: 16) ?? user.GetDefaultAvatarUrl()) is { } avatarUrl)
+                {
+                    aviColor = _imageService.GetDominantColorAsync(new Uri(avatarUrl));
+                }
+                else
+                {
+                    aviColor = new ValueTask<Color>(Color.Teal);
+                }
+
+                var embed = new EmbedBuilder()
+                    .WithTitle($"#{campaign.Id}")
+                    .WithDescription($"**{campaign.Subject.Nickname}** has been nominated for promotion to {boldRole} ")
```

Comment:
This looks weird if the user doesn't have a nickname set:
![image](https://user-images.githubusercontent.com/2829282/77378194-b41d1100-6d4b-11ea-8c4d-8b19bc4b0ad0.png)


```suggestion
                    .WithDescription($"**{campaign.Subject.GetFullUsername()}** has been nominated for promotion to {boldRole} ")
```

Reply:
Whatever happened to our property/extension that used `.Nickname` but would fallback to `{.Username}#{.Discriminator}` when `.Nickname` is `null`?

Reply:
https://github.com/discord-csharp/MODiX/pull/402